roles/kalite/tasks/install.yml

```diff
@@ -17,9 +17,13 @@
       - python-setuptools    # Provides setuptools-44 on recent OS's (last version compatible with python2)
       - virtualenv           # Drags in 'python3-virtualenv' which in turn drags in 'python3-pip' -- for Ansible module 'pip' when used with 'virtualenv_command: /usr/bin/virtualenv' and 'virtualenv_python: python2.7' -- compare package 'python3-venv' used by roles {calibre-web, jupyterhub, lokole}
     state: present
-  #when: not (is_debian_9 or is_debian_10 or is_ubuntu_16 or is_ubuntu_17 or is_ubuntu_18 or is_ubuntu_19)
+  when: not (is_debian_12 or is_ubuntu_2304)
   # 2020-03-31: Testing for {is_raspbian_9, is_raspbian_10} is not currently nec, as testing for {is_debian_9, is_debian_10} covers that already.
 
+- name: Use scripts/install_python2.sh to install python2 and virtualenv
+  command: "{{ iiab_dir }}/scripts/install_python2.sh"
+  when: is_debian_12 or is_ubuntu_2304
```

Comment:
```suggestion
  when: is_debian and os_ver is version('debian-12', '>=') or is_ubuntu and os_ver is version('ubuntu-2304', '>=')
```

Reply:
can't we just drop the is_debian is_ubuntu part? In the end os_ver ends up defining those through OS.yml

Reply:
Oops, let me take that back.  It's not good enough, given:
- `is_debian` also permits `os_ver: rasbian-11`
- `is_ubuntu` also permits `os_ver: linuxmint-21`

Possibly we should act on the output of a command like `apt list python2` (instead of hard-coding any OS versions).

Reply:
this would be excluding raspbian-12 that would have is_debian_12 defined in the OS.yml file
```
 104 [runtime]
 105 iiab_stage = 9
 106 iiab_base_ver = 8.1
 107 iiab_revision = 0
 108 iiab_remote_url = none
 109 runtime_branch = python2_fix
 110 runtime_commit = 6e61e55948f9254b4da9dff7b3ad457cdc103945
 111 iiab_recent_tag = 8.1-preview-1
 112 runtime_date = 2023-04-06T01:52:53Z
 113 ansible_version = 2.14.4
 114 kernel = 6.1.19-v8+
 115 memory_mb = 3794
 116 swap_mb = 1023
 117 rpi_model = Raspberry Pi 4 Model B Rev 1.1
 118 devicetree_model = Raspberry Pi 4 Model B Rev 1.1
 119 os_ver = raspbian-12
 120 python_version = 3.11
 121 php_version = 8.2
```

Reply:
> Possibly we should act on the output of a command like `apt list python2` (instead of hard-coding any OS versions).

The known part is which distros don't have python2, the jury is out on what RasPiOS is going to ship for 'version 12' but I have a small preview with arm64.